packages/middleware-flexible-checksums/src/flexibleChecksumsMiddleware.ts

```diff
@@ -54,15 +54,25 @@ export const flexibleChecksumsMiddleware =
       return next(args);
     }
 
-    if (hasHeaderWithPrefix("x-amz-checksum-", args.request.headers)) {
+    const { request, input } = args;
+    const { body: requestBody, headers } = request;
+    const { requestChecksumRequired, requestAlgorithmMember } = middlewareConfig;
+
+    if (hasHeaderWithPrefix("x-amz-checksum-", headers)) {
+      // Remove input[requestAlgorithmMember] and header, if it was added by flexibleChecksumsInputMiddleware
```

Comment:
this should state the condition more completely, since 

> Remove input[requestAlgorithmMember] and header, if it was added by flexibleChecksumsInputMiddleware

sounds like an unconditional removal 

it should perhaps say "... and the user has set the checksum via a modeled field"?